prow/deck/jobs/jobs.go

```diff
@@ -80,15 +84,75 @@ type PodLogClient interface {
 	GetLogs(name string, opts *coreapi.PodLogOptions) ([]byte, error)
 }
 
+// PJListingClient is an interface to list ProwJobs
+type PJListingClient interface {
+	List(context.Context, *prowapi.ProwJobList, ...ctrlruntimeclient.ListOption) error
+}
+
 // NewJobAgent is a JobAgent constructor.
-func NewJobAgent(kc serviceClusterClient, plClients map[string]PodLogClient, cfg config.Getter) *JobAgent {
+func NewJobAgent(ctx context.Context, pjLister PJListingClient, hiddenOnly, showHidden bool, plClients map[string]PodLogClient, cfg config.Getter) *JobAgent {
 	return &JobAgent{
-		kc:     kc,
+		kc: &filteringProwJobLister{
+			ctx:         ctx,
+			client:      pjLister,
+			hiddenRepos: func() sets.String { return sets.NewString(cfg().Deck.HiddenRepos...) },
+			hiddenOnly:  hiddenOnly,
+			showHidden:  showHidden,
+		},
 		pkcs:   plClients,
 		config: cfg,
 	}
 }
 
+type filteringProwJobLister struct {
+	ctx         context.Context
+	client      PJListingClient
+	cfg         config.Getter
+	hiddenRepos func() sets.String
+	hiddenOnly  bool
+	showHidden  bool
+}
+
+func (c *filteringProwJobLister) ListProwJobs(selector string) ([]prowapi.ProwJob, error) {
+	prowJobList := &prowapi.ProwJobList{}
+	parsedSelector, err := labels.Parse(selector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse selector: %v", err)
+	}
+	listOpts := &ctrlruntimeclient.ListOptions{LabelSelector: parsedSelector, Namespace: c.cfg().ProwJobNamespace}
```

Comment:
Including the namespace here is the only actual change made by this PR. Everything else is just moving code around (Git can unfortunately not show that unless its the whole file).